engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -24,66 +25,107 @@
   Net::HTTPRetriableError
 ].freeze
 
+INSTANCE_ID_KEYS = {
+  Amazon: 'instanceId',
+  Google: 'instance_id',
+  Microsoft: 'vmId'
+}.freeze
+
 # rubocop:disable Metrics/ModuleLength
 module SccProxy
   class << self
 
-    def headers(auth)
+    # rubocop:disable ThreadSafety/ClassAndModuleAttributes
+    attr_accessor :instance_id
+
+    # rubocop:enable ThreadSafety/ClassAndModuleAttributes
+
+    # rubocop:disable ThreadSafety/InstanceVariableInClassMethod
+    def headers(auth, params, logger)
+      @instance_id = if params && params.class != String
+                       get_instance_id(params, logger)
+                     else
+                       # if it is not JSON, it is the system_token already
+                       params
+                     end
+
       {
         'accept' => 'application/json,application/vnd.scc.suse.com.v4+json',
         'Content-Type' => 'application/json',
-        'Authorization' => auth
+        'Authorization' => auth,
+        ApplicationController::SYSTEM_TOKEN_HEADER => @instance_id
       }
     end
+    # rubocop:enable ThreadSafety/InstanceVariableInClassMethod
+
+    def get_instance_id(params, logger)
```

Comment:
I would love to get rid of `logger` here. I do not really see why its needed, since the `provider` is only initialized half? What is the reason to use it at all here?